### PR TITLE
Add Dart SDK constraint to example pubspecs that were missing them

### PR DESCRIPTION
**Pull request by zanderso**
If a pubspec.yaml for an example was missing a Dart SDK constraint, this PR copies the `environment` section from the `pubspec.yaml` file in the parent directory.

### CODE REVIEW

**Review by leafpetersen — APPROVED** [review]

**Comment by amirh** [comment]
We should probably re-publish all these plugins (to make sure the pub.dev score won't get hit by analysis errors) so we need to bump the version numbers and add changelog entries.

Side note: it is worrying that so many packages had this problem and were affected by https://github.com/flutter/flutter/pull/69440#issuecomment-719887060, I'm wondering what's the status for all the existing packages in the wild (is it possible that we had a time range during which the plugin template didn't include these constraints?)

**Comment by leafpetersen** [comment]
> Side note: it is worrying that so many packages had this problem and were affected by [flutter/flutter#69440 (comment)](https://github.com/flutter/flutter/pull/69440#issuecomment-719887060), I'm wondering what's the status for all the existing packages in the wild (is it possible that we had a time range during which the plugin template didn't include these constraints?)

I think @mit-mit would know about the template, and might have run the numbers for missing sdk (or at least flutter) constraints once?

**Comment by zanderso** [comment]
> we need to bump the version numbers and add changelog entries.

Done

**Review by amirh — APPROVED** [review]
LGTM

File: packages/android_alarm_manager/example/pubspec.yaml [hunk 1]
```diff
@@ -21,3 +21,7 @@ dev_dependencies:
 
 flutter:
   uses-material-design: true
+
+environment:
+  sdk: ">=2.1.0 <3.0.0"
+  flutter: ">=1.12.13+hotfix.5 <2.0.0"
```
Comment (mit-mit) [inline packages/android_alarm_manager/example/pubspec.yaml:27]: For the Flutter SDK constraint, please add the lower constraint only. See https://github.com/flutter/plugins/pull/3236 and https://github.com/flutter/flutter/pull/68437

**Review by mit-mit — CHANGES_REQUESTED** [review]

File: packages/android_alarm_manager/example/pubspec.yaml (hunk 1 above)
Comment (zanderso) [reply packages/android_alarm_manager/example/pubspec.yaml:27]: The goal of this PR is to get flutter/plugins CI green with tip-of-tree framework. Other cleanup work can happen in a subsequent PR.

Comment (mit-mit) [reply packages/android_alarm_manager/example/pubspec.yaml:27]: Why would we add a constraint in this PR, and then remove it again later? I don't understand the value in incurring additional debt. The `flutter` constraint does nothing wrt. getting CI green; you just need the Dart constraint for that.

Comment (zanderso) [reply packages/android_alarm_manager/example/pubspec.yaml:27]: This PR is consistent with the rest of the repo. Fixing every pubspec in the repo is out of scope for this PR. Choosing the right time to do other cleanups is up to @amirh to decide.

**Comment by zanderso** [comment]
@amirh the Cirrus presubmits are timing out. Do you know what's up with that?

**Comment by amirh** [comment]
> @amirh the Cirrus presubmits are timing out. Do you know what's up with that?

Seems to be https://github.com/flutter/flutter/issues/69439 😞

**Comment by amirh** [comment]
This has been incorporated into https://github.com/flutter/plugins/pull/3239 which also works around the driver timeout.

**Comment by zanderso** [comment]
Closing in favor of https://github.com/flutter/plugins/pull/3239

**Pull request closed without merging**